scripts/generate_special_function_declarations.py

```diff
@@ -84,14 +84,14 @@ def parse_capsule_name(capsule):
     # There isn't a Python equivalent to `PyCapsule_GetName`, so
     # resort to a hacky method for finding the signature.
     match = re.match(
-        '\<capsule object "(?P<signature>.+)" at 0x[a-f0-9]+\>',
+        '<capsule object "(?P<signature>.+)" at 0x[a-f0-9]+>',
         str(capsule),
     )
     if match is None:
         raise ValueError('Unexpected capsule name {}'.format(capsule))
 
     signature = match.group('signature')
-    match = re.match('(?P<return_type>.+) \((?P<arg_types>.+)\)', signature)
+    match = re.match('(?P<return_type>.+) ((?P<arg_types>.+))', signature)
```

Comment:
This change may have to be reverted as it may potentially break generating the docs.

Reply:
Yes, if you want to match literal parentheses (which seems to be the case here) you definitely need that escape. It would also be good form to use a raw string literal here
```suggestion
    match = re.match(r'(?P<return_type>.+) \((?P<arg_types>.+)\)', signature)
```
If I'm reading the current version correctly, the parentheses created another capture group, which would mean that the `<arg_types>` named group still exists, but now it includes the enclosing parentheses as well.

But it was further broken because the first `.+` in `<return_type>` could match the whitespace as well, greedily eating the whole string into `<return_type>`!
```py
>>> re.match('(?P<return_type>.+) ((?P<arg_types>.+))', 'double (double, double, double, int __pyx_skip_dispatch)').group('return_type')
'double (double, double, double, int'
```
The first part greedily matched until the last space.

If we add back the matching for the literal parentheses that will stop this behaviour (without having to make the first group's mattern more complicated):
```py
>>> re.match(r'(?P<return_type>.+) \((?P<arg_types>.+)\)', 'double (double, double, double, int __pyx_skip_dispatch)').group('return_type')
'double'
```

Reply:
💯 i concur

Reply:
What is the advantage of using a raw string literal here?

Reply:
If you use `'\<'` that tries to be a python string escape (think `'\n'` and `'\t'`). Since `'\<'` is not a valid escape python (for the time being) ignores it and creates the equivalent of `r'\<'`: a string of length 2 where the first character is a backslash and the second is `<`.

There was a plan to make python warn and eventually raise when it encountered such invalid escapes, but it was reverted (see [3.6 release notes](https://docs.python.org/3/whatsnew/3.6.html#deprecated-python-behavior)). Semantically speaking `'\<'` is nonsense. In a regex you always want literal backslashes in the string itself so that the regex engine can decide what it makes of it. But for now the two things are equivalent:
```py
>>> '\<'
'\\<'
>>> r'\<'
'\\<'
```

Reply:
And of course the same goes for `'\('` (I was just in the mindset of the other similar comment). So the raw string literal is the only correct choice here. That or using `'\\'` in the string (but that's what raw string literals are for).